### PR TITLE
Added sound_new and sound_init functions

### LINKED ISSUE #1622: sound: Add initializing functions for sound type declaration
This issue is created because we found that there are no initialization functions of any kind for `sound_t` module, which comes to be a huge hassle while writing for tests, because the same lines of code will have to be repeated (specifically, `malloc` of `sound_t` and assigning values and pointers within `sound_t`.

### PR DESCRIPTION
**Pull request by jiaoboxuan**
Summary: Existent functions such as `load_wav` and `sound_free` require already allocated `sound_t` structs, and while writing tests, I would have to allocate the structs in separate lines of code, and repeat the process per new test. Therefore, I added two functions, `sound_init` and `sound_new`, descriptions of which are added in `sound.h` file. The code compiles successfully in the workflow run, and these two simple functions could significantly ease the test writing process, as well as improve the readability of the module.

### CODE REVIEW

File: include/sound/sound.h [hunk 1]
```diff
@@ -31,12 +31,34 @@ typedef struct {
...
+
+/* Initializes the 
+ * 
+ * Parameters:
+ *      None
+ * Return:
+ *      An allocated sound structure, with name and type
+ */
+int sound_init(sound_t *sound, SoundType type, char *name);
+
```
Comment (JaStarbuck) [inline include/sound/sound.h:63]: I like your documentation here!

File: src/sound/src/sound.c [hunk 2]
```diff
@@ -1,6 +1,26 @@
 #include "common/uthash.h"
 #include "sound/sound.h"
 
+/* See sound.h for details */
+sound_t *sound_new(SoundType type, char *name)
+{
+    sound_t *sound = (sound_t*)malloc(sizeof(sound_t));
+    sound->wavBuffer = (uint8_t*)malloc(sizeof(uint8_t));
```
Comment (JaStarbuck) [inline src/sound/src/sound.c:8]: not sure if this needs to be malloced

**Review by JaStarbuck — APPROVED** [review]
This looks really good! I've gone through and left comments on some things. I wonder if once you malloc the struct, you actually have to malloc the things inside the struct. Other than that, it looks good!

**Review by JaStarbuck — APPROVED** [review]
Oops think I accidentally re-review--approving again

File: include/sound/sound.h [hunk 3]
```diff
@@ -31,12 +31,34 @@ typedef struct {
...
+ *      type: enum SoundType
+ *      name: name of the sound, 
+ *            or more specifically, the path to a sound file.
+ *            Must be already allocated in the heap
+ * Return:
+ *      An allocated sound structure, with name and type
+ */
+sound_t *sound_new(SoundType type, char *name);
+
+/* Initializes the
```
Comment (ecoble) [inline include/sound/sound.h]: Initializes the what? Also it looks like function does have parameters, and it is returning an int.

File: src/sound/src/sound.c (hunk 2 above)
Comment (ecoble) [reply src/sound/src/sound.c:8]: Mallocing this is correct since wavBuffer is a pointer.

**Review by ecoble — CHANGES_REQUESTED** [review]
Nice work! Just a quick change on the header comment.

**Comment by jiaoboxuan** [comment]
Update: changed the `char` structure from pointer form to stack form, and updated the documentation.

**Review by ecoble — APPROVED** [review]

**Comment by jwschroeder21** [comment]
Closing statement: sound_init and sound_new were added with descriptions of them in the sound.h file. These functions ease the test writing process and allow  for no re-allocation of the sound_t structs in each process of a repeated test.

**Comment by ecoble** [comment]
Issue Score: Excellent

Comments:
Great work! Nice job identifying the need for new and init functions!

**Pull request merged**